src/components/Footer/Footer.js

```diff
@@ -28,8 +39,21 @@ const Footer = ({
   blockStatus,
   loadingVote,
   showScoreButton = false,
-  onPressScore
+  onPressScore,
+  isShowDM = false
 }) => {
+  const {sendMessageDM} = useDMMessage();
+  const [profile] = React.useContext(Context).profile;
+  const [userAllowDm, setUserAllowDm] = React.useState(false);
+
+  const [loading, setLoading] = React.useState({
+    loadingDm: false,
+    loadingDmAnon: false,
+    loadingGetAllowAnonDmStatus: false
+  });
+  const refSheet = React.useRef();
+  const {createSignChat} = useCreateChat();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The state management in this component seems to be getting complex with multiple states being managed. Consider using a reducer (useReducer) for better state management and readability.
```diff
-  const [userAllowDm, setUserAllowDm] = React.useState(false);
-  const [loading, setLoading] = React.useState({
-    loadingDm: false,
-    loadingDmAnon: false,
-    loadingGetAllowAnonDmStatus: false
-  });

+  const initialState = {
+    userAllowDm: false,
+    loadingDm: false,
+    loadingDmAnon: false,
+    loadingGetAllowAnonDmStatus: false
+  };
+
+  function reducer(state, action) {
+    switch (action.type) {
+      case 'SET_USER_ALLOW_DM':
+        return {...state, userAllowDm: action.payload};
+      case 'SET_LOADING':
+        return {...state, ...action.payload};
+      default:
+        throw new Error();
+    }
+  }
+
+  const [state, dispatch] = React.useReducer(reducer, initialState);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->